diff -Nurb streamdev-0.5.0-pre-20090706/Makefile streamdev-0.5.0-pre-20090706-vdr1.7/Makefile
--- streamdev-0.5.0-pre-20090706/Makefile	2009-07-01 12:46:15.000000000 +0200
+++ streamdev-0.5.0-pre-20090706-vdr1.7/Makefile	2010-02-07 06:35:46.164272873 +0100
@@ -42,6 +42,7 @@
 INCLUDES += -I$(VDRDIR)/include -I.
 
 DEFINES += -D_GNU_SOURCE -DPLUGIN_NAME_I18N='"$(PLUGIN)"'
+DEFINES += -D_FILE_OFFSET_BITS=64 -D_LARGEFILE_SOURCE -D_LARGEFILE64_SOURCE
 
 ### The object files (add further files here):
 
