--- a/touchegg-gui/src/touchegg-gui/data/transaction_manager/TransactionManager.cpp
+++ b/touchegg-gui/src/touchegg-gui/data/transaction_manager/TransactionManager.cpp
@@ -19,10 +19,10 @@
 // **********             STATIC METHODS AND VARIABLES             ********** //
 // ************************************************************************** //
 
-const char* TransactionManager::HOME_CONFIG_FILE ="/.touchegg/touchegg.conf";
+const char* TransactionManager::HOME_CONFIG_FILE ="/.config/touchegg/touchegg.conf";
 const char* TransactionManager::USR_SHARE_CONFIG_FILE =
         "/usr/share/touchegg/touchegg.conf";
-const char* TransactionManager::HOME_CONFIG_DIR = ".touchegg";
+const char* TransactionManager::HOME_CONFIG_DIR = ".config/touchegg";
 
 //------------------------------------------------------------------------------
 
