--- a/src/touchegg-gui/presentation/config_forms/ConfigForm.h
+++ b/src/touchegg-gui/presentation/config_forms/ConfigForm.h
@@ -16,6 +16,7 @@
 #define CONFIGFORM_H
 
 #include "src/touchegg-gui/util/Include.h"
+#include <QDialog>
 
 /**
  * @~spanish
--- a/touchegg-gui.pro
+++ b/touchegg-gui.pro
@@ -1,6 +1,6 @@
 TEMPLATE        = app
 TARGET          = touchegg-gui
-QT             += core gui xml
+QT             += core gui widgets xml x11extras
 OBJECTS_DIR    += build
 MOC_DIR        += build
 RCC_DIR        += build
--- a/src/touchegg-gui/presentation/widgets/MenuItem.h
+++ b/src/touchegg-gui/presentation/widgets/MenuItem.h
@@ -16,6 +16,8 @@
 #define MENUITEM_H
 
 #include "src/touchegg-gui/util/Include.h"
+#include <QWidget>
+
 namespace Ui { class MenuItem; }
 
 /**
--- a/src/touchegg-gui/presentation/widgets/GestureConfigWidget.h
+++ b/src/touchegg-gui/presentation/widgets/GestureConfigWidget.h
@@ -19,6 +19,8 @@
 #include "src/touchegg-gui/presentation/config_forms/ConfigFormFactory.h"
 #include "src/touchegg-gui/presentation/config_forms/ConfigForm.h"
 #include "src/touchegg-gui/logic/facade/Facade.h"
+#include <QFrame>
+
 namespace Ui { class GestureConfigWidget; }
 
 
--- a/src/touchegg-gui/presentation/main_window/MainWindow.h
+++ b/src/touchegg-gui/presentation/main_window/MainWindow.h
@@ -19,6 +19,8 @@
 #include "src/touchegg-gui/presentation/widgets/MenuItem.h"
 #include "src/touchegg-gui/presentation/main_window/GesturesTab.h"
 #include "src/touchegg-gui/presentation/main_window/GeneralSettingsTab.h"
+#include <QMainWindow>
+
 namespace Ui { class MainWindow; }
 
 /**
--- a/src/touchegg-gui/presentation/main_window/GeneralSettingsTab.h
+++ b/src/touchegg-gui/presentation/main_window/GeneralSettingsTab.h
@@ -17,6 +17,8 @@
 
 #include "src/touchegg-gui/util/Include.h"
 #include "src/touchegg-gui/logic/facade/Facade.h"
+#include <QFrame>
+
 namespace Ui { class GeneralSettingsTab; }
 
 
--- a/src/touchegg-gui/presentation/main_window/GesturesTab.h
+++ b/src/touchegg-gui/presentation/main_window/GesturesTab.h
@@ -18,6 +18,9 @@
 #include "src/touchegg-gui/util/Include.h"
 #include "src/touchegg-gui/logic/facade/Facade.h"
 #include "src/touchegg-gui/presentation/widgets/GestureConfigWidget.h"
+#include <QFrame>
+#include <QLineEdit>
+
 namespace Ui { class GesturesTab; }
 
 
--- a/src/touchegg-gui/Main.cpp
+++ b/src/touchegg-gui/Main.cpp
@@ -13,6 +13,7 @@
  */
 #include "src/touchegg-gui/util/Include.h"
 #include "src/touchegg-gui/presentation/main_window/MainWindow.h"
+#include <QApplication>
 
 
 /**
