Index: suricata-2.0.8/configure
===================================================================
--- suricata-2.0.8.orig/configure
+++ suricata-2.0.8/configure
@@ -14314,7 +14314,7 @@ $as_echo_n "checking host os... " >&6; }
             LDFLAGS="${LDFLAGS} -L/opt/local/lib"
             ;;
         *-*-linux*)
-            #for now do nothing
+	    e_magic_file="/usr/share/misc/magic.mgc"
             ;;
         *-*-mingw32*)
             CFLAGS="${CFLAGS} -DOS_WIN32"
