Index: trunk/xbmc/cores/paplayer/ModuleCodec/Makefile.in
===================================================================
--- trunk/xbmc/cores/paplayer/ModuleCodec/Makefile.in (revision 22665)
+++ trunk/xbmc/cores/paplayer/ModuleCodec/Makefile.in (revision 24408)
@@ -25,5 +25,5 @@
 
 dumb/lib/unix/libdumb.a: compile
-	$(MAKE) -C dumb CC="$(CC)" CFLAGS="$(CFLAGS) -Iinclude -Wall -W -Wwrite-strings -Wstrict-prototypes -Wmissing-declarations -DDUMB_DECLARE_DEPRECATED -O2 -ffast-math -fomit-frame-pointer"
+	$(MAKE) -C dumb CC="$(CC)" CFLAGS="$(CFLAGS) -Iinclude -DDUMB_DECLARE_DEPRECATED"
 
 include ../../../../Makefile.include
