Index: trunk/xbmc/cores/ExternalPlayer/ExternalPlayer.cpp
===================================================================
--- trunk/xbmc/cores/ExternalPlayer/ExternalPlayer.cpp (revision 23602)
+++ trunk/xbmc/cores/ExternalPlayer/ExternalPlayer.cpp (revision 24382)
@@ -40,4 +40,5 @@
 #include "XMLUtils.h"
 #include "utils/TimeUtils.h"
+#include "utils/log.h"
 #if defined(_WIN32)
   #include "Windows.h"
