Index: xbmc/lib/cximage-6.0/CxImage/ximajpg.cpp
===================================================================
--- xbmc/lib/cximage-6.0/CxImage/ximajpg.cpp	(revision 26653)
+++ xbmc/lib/cximage-6.0/CxImage/ximajpg.cpp	(working copy)
@@ -220,6 +220,7 @@
 
 	// Set the scale <ignacio>
 	cinfo.scale_denom = GetJpegScale();
+	cinfo.scale_num = 1;
 
 	// Borrowed the idea from GIF implementation <ignacio>
 	if (info.nEscape == -1) {
