--- a/src/qt/clientmodel.cpp
+++ b/src/qt/clientmodel.cpp
@@ -9,6 +9,8 @@
 #include <qt/guiutil.h>
 #include <qt/peertablemodel.h>
 
+#include <boost/bind.hpp>
+
 #include <chain.h>
 #include <chainparams.h>
 #include <checkpoints.h>
--- a/src/qt/nixgui.h
+++ b/src/qt/nixgui.h
@@ -11,6 +11,8 @@
 
 #include <amount.h>
 
+#include <boost/bind.hpp>
+
 #include <QLabel>
 #include <QMainWindow>
 #include <QMap>
--- a/src/validationinterface.cpp
+++ b/src/validationinterface.cpp
@@ -17,6 +17,7 @@
 #include <atomic>
 #include <future>
 
+#include <boost/bind.hpp>
 #include <boost/signals2/signal.hpp>
 
 struct MainSignalsInstance {
