Index: sgminer-5.0.1/configure.ac
===================================================================
--- sgminer-5.0.1.orig/configure.ac
+++ sgminer-5.0.1/configure.ac
@@ -245,8 +245,7 @@ AM_CONDITIONAL([HAVE_CURSES], [test x$cu
 AM_CONDITIONAL([HAVE_WINDOWS], [test x$have_win32 = xtrue])
 AM_CONDITIONAL([HAVE_x86_64], [test x$have_x86_64 = xtrue])
 
-AC_CONFIG_SUBDIRS([submodules/jansson])
-JANSSON_LIBS="submodules/jansson/src/.libs/libjansson.a"
+JANSSON_LIBS="-ljansson"
 
 PKG_PROG_PKG_CONFIG()
 
@@ -365,7 +364,6 @@ AC_SUBST(ADL_CPPFLAGS)
 
 AC_CONFIG_FILES([
 	Makefile
-	submodules/Makefile
 	ccan/Makefile
 	lib/Makefile
 	sph/Makefile
Index: sgminer-5.0.1/Makefile.am
===================================================================
--- sgminer-5.0.1.orig/Makefile.am
+++ sgminer-5.0.1/Makefile.am
@@ -1,18 +1,16 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-JANSSON_CPPFLAGS= -I$(top_builddir)/submodules/jansson/src -I$(top_srcdir)/submodules/jansson/src
-
 EXTRA_DIST	= example.conf m4/gnulib-cache.m4 \
 		  ADL_SDK/readme.txt api-example.php miner.php	\
 		  API.class API.java api-example.c hexdump.c \
 		  doc/API doc/FAQ doc/GPU doc/SCRYPT doc/windows-build.txt
 
-SUBDIRS		= lib submodules ccan sph
+SUBDIRS		= lib ccan sph
 
 bin_PROGRAMS     = sgminer
 
-sgminer_CPPFLAGS = $(PTHREAD_FLAGS) -fno-strict-aliasing $(JANSSON_CPPFLAGS)
+sgminer_CPPFLAGS = $(PTHREAD_FLAGS) -fno-strict-aliasing
 sgminer_LDFLAGS  = $(PTHREAD_FLAGS)
 sgminer_LDADD    = $(DLOPEN_FLAGS) @LIBCURL_LIBS@ @JANSSON_LIBS@ @PTHREAD_LIBS@ \
 		  @OPENCL_LIBS@ @NCURSES_LIBS@ @PDCURSES_LIBS@ @WS2_LIBS@ \
