Index: support/configure.inc
===================================================================
--- support/configure.inc	(Revision 3830)
+++ support/configure.inc	(Arbeitskopie)
@@ -120,41 +120,7 @@
 
 
 setup_ffmpeg() {
-    echo "Configuring FFmpeg"
-    mkdir -p ${BUILDDIR}/ffmpeg/build
-    rm -rf ${BUILDDIR}/ffmpeg/install
-
-    (cd ${BUILDDIR}/ffmpeg/build && \
-	${TOPDIR}/ext/ffmpeg/configure  \
-	$CC_FFMPEG \
-	--disable-encoders \
-	--enable-encoder=mp2 \
-	--enable-encoder=mpeg2video \
-	--disable-muxers \
-	--enable-muxer=matroska \
-	--disable-decoders \
-	--enable-decoder=mp2 \
-	--enable-decoder=mp3 \
-	--enable-decoder=mpeg2video \
-	--enable-decoder=h264 \
-	--enable-decoder=ac3 \
-	--disable-demuxers \
-	--disable-filters \
-	--disable-devices \
-	--disable-stripping \
-	--enable-static \
-	--disable-shared \
-	--enable-pthreads \
-	--prefix=${BUILDDIR}/ffmpeg/install \
-	--disable-ffserver \
-	--disable-ffmpeg \
-	--disable-ffplay \
-	$@
-	)
-
 cat >> ${CONFIG_MAK} << EOF 
-CFLAGS_cfg  += -I${BUILDDIR}/ffmpeg/install/include
-LDFLAGS_cfg += -L${BUILDDIR}/ffmpeg/install/lib
 LDFLAGS_cfg += -lavformat -lavcodec -lavutil
 EOF
 }
@@ -179,4 +145,4 @@
     echo "Final binary is: build.${PLATFORM}/tvheadend"
 
     die
-}
\ No newline at end of file
+}
Index: Makefile
===================================================================
--- Makefile	(Revision 3830)
+++ Makefile	(Arbeitskopie)
@@ -142,21 +142,17 @@
 
 all:	${PROG}
 
-.PHONY:	clean distclean ffmpeg
+.PHONY:	clean distclean
 
-${PROG}: ${BUILDDIR}/ffmpeg/install $(OBJDIRS) $(OBJS) $(BUNDLE_OBJS) Makefile
+${PROG}: $(OBJDIRS) $(OBJS) $(BUNDLE_OBJS) Makefile
 	$(CC) -o $@ $(OBJS) $(BUNDLE_OBJS) $(LDFLAGS) ${LDFLAGS_cfg}
 
 $(OBJDIRS):
 	@mkdir -p $@
 
-${BUILDDIR}/%.o: %.c  ${BUILDDIR}/ffmpeg/install
+${BUILDDIR}/%.o: %.c
 	$(CC) -MD $(CFLAGS_com) $(CFLAGS) $(CFLAGS_cfg) -c -o $@ $(CURDIR)/$<
 
-${BUILDDIR}/ffmpeg/install ffmpeg:
-	cd ${BUILDDIR}/ffmpeg/build && ${MAKE} all
-	cd ${BUILDDIR}/ffmpeg/build && ${MAKE} install
-
 clean:
 	rm -rf ${BUILDDIR}/src ${BUILDDIR}/bundles
 	find . -name "*~" | xargs rm -f
