Index: libiscsi-1.12.0/examples/Makefile.am
===================================================================
--- libiscsi-1.12.0.orig/examples/Makefile.am
+++ libiscsi-1.12.0/examples/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS=-I. -I../include "-D_U_=__attribute__((unused))" \
+AM_CPPFLAGS=-I. -I${srcdir}/../include "-D_U_=__attribute__((unused))" \
 	"-D_R_(A,B)=__attribute__((format(printf,A,B)))"
 AM_CFLAGS=$(WARN_CFLAGS)
 LDADD = ../lib/libiscsi.la
Index: libiscsi-1.12.0/lib/Makefile.am
===================================================================
--- libiscsi-1.12.0.orig/lib/Makefile.am
+++ libiscsi-1.12.0/lib/Makefile.am
@@ -17,7 +17,7 @@ libiscsi_la_LDFLAGS = \
 	-version-info $(SOCURRENT):$(SOREVISON):$(SOAGE) -bindir $(bindir) \
 	-no-undefined -export-symbols libiscsi.syms
 
-libiscsi_la_CPPFLAGS = -I../include -I$(srcdir)/include \
+libiscsi_la_CPPFLAGS = -I../include -I$(srcdir)/../include \
 	"-D_U_=__attribute__((unused))" \
 	"-D_R_(A,B)=__attribute__((format(printf,A,B)))"
 
Index: libiscsi-1.12.0/utils/Makefile.am
===================================================================
--- libiscsi-1.12.0.orig/utils/Makefile.am
+++ libiscsi-1.12.0/utils/Makefile.am
@@ -1,4 +1,4 @@
-AM_CPPFLAGS=-I../include "-D_U_=__attribute__((unused))" \
+AM_CPPFLAGS=-I${srcdir}/../include "-D_U_=__attribute__((unused))" \
 	"-D_R_(A,B)=__attribute__((format(printf,A,B)))"
 AM_CFLAGS=$(WARN_CFLAGS)
 LDADD = ../lib/libiscsi.la
